packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dq-response.ts

```diff
@@ -102,16 +141,42 @@ export function parseDocumentReference({
 
     const slot = classificationSlots.find((s: Slot) => s._name === slotName);
     return getSlotValue(slot);
-  };
+  }
 
-  const findClassificationName = (scheme: string): string | undefined => {
+  function findClassificationName(scheme: string): string | undefined {
     const classification = classifications?.find(
       (classification: Classification) => classification?._classificationScheme === scheme
     );
     if (!classification) return undefined;
     const title = getNameValue(classification?.Name);
     return title;
-  };
+  }
+
+  function findClassificationCoding(classificationScheme: string): Coding | undefined {
+    const classification = classifications.find(
+      (c: Classification) => c._classificationScheme === classificationScheme
+    );
+    if (!classification) return undefined;
+
+    const code = classification._nodeRepresentation;
+    const display = getNameValue(classification.Name);
+    const system = getCodeSystemForScheme(classificationScheme);
+
+    if (!code && !display) return undefined;
+
+    const result = codingSchema.safeParse({
+      system,
+      code,
+      display,
+    });
+
+    if (!result.success) {
+      log(`Error parsing coding for classification ${classificationScheme}`);
```

Comment:
nit: don't say it's an error if we're failing gracefully, maybe "Could not parse coding..."?